inc/Classes/DB.php

```diff
@@ -110,11 +110,13 @@ public function connect($save = false)
         $user = $config['database']['user'];
         $pass = $config['database']['passwd'];
         $database = $config['database']['database'];
+        $charset = $config['database']['dbport'];
         $charset = $config['database']['charset'];
+	    
 
         // Try to connect to the database
         // Suppress error output, because mysqli_connect throws a PHP Warning once it is not able to connect
-        $this->link_id = @mysqli_connect($server, $user, $pass);
+        $this->link_id = @mysqli_connect($server, $user, $pass, $database, $dbport);
```

Comment:
Whilst the code directly below selects the DB as next action I would also be in favour to do so directly on connect.
Not specifying the DB would require the DB user to have `USAGE ON *.*` as privilege, if I'm not mistaken. 
Which _may_ be an issue in shared environments, but _should_ be not.

Just not sure if resulting code does handle DB-selection related errors OK.
Was that tested?